server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -209,6 +163,54 @@ public int compareTo(DocIdToIndex o) {
         }
     }
 
+    private FieldsVisitor createStoredFieldsVisitor(SearchContext context, Map<String, Set<String>> storedToRequestedFields) {
```

Comment:
do we want to make this function static?

Reply:
👍

Reply:
On second thought, would you be okay keeping these as instance methods for consistency? It looks like all the other helper methods (`findRootDocumentIfNested`, `prepareHitContext`, etc.) are not static.

Reply:
I don't mind keeping them instance methods for consistency.